packages/broker/src/plugins/metrics/MetricsPlugin.ts

```diff
@@ -1,16 +1,99 @@
+import { MetricsContext } from 'streamr-network'
 import { router as volumeEndpoint } from './VolumeEndpoint'
 import { Plugin, PluginOptions } from '../../Plugin'
+import PLUGIN_CONFIG_SCHEMA from './config.schema.json'
+import { StorageNodeRegistryItem } from '../../config'
+import { VolumeLogger } from './VolumeLogger'
+import { StreamrClient } from 'streamr-client'
+import { Schema } from 'ajv'
 
-export class MetricsPlugin extends Plugin<void> {
+export interface MetricsPluginConfig {
+    consoleLogIntervalInSeconds: number
+    legacyMetricsStreamId: string | null
+    clientWsUrl?: string
+    clientHttpUrl?: string
+    perNodeMetrics: {
+        enabled: boolean
+        intervals: {
+            sec: number,
+            min: number,
+            hour: number,
+            day: number
+        } | null
+        storageNode: string
+    } | null
+}
+
+export class MetricsPlugin extends Plugin<MetricsPluginConfig> {
+    private readonly volumeLogger: VolumeLogger
 
     constructor(options: PluginOptions) {
         super(options)
+        this.volumeLogger = createVolumeLogger(
+            this.brokerConfig.ethereumPrivateKey,
+            this.pluginConfig,
+            this.metricsContext,
+            this.nodeId,
+            this.storageNodeRegistry.getStorageNodes()
+        )
+    }
+
+    async start(): Promise<unknown> {
```

Comment:
In `publishHttp` plugin etc. there is no conditionality for httpServer but the Broker will throw an error if there is no config:
```
if (config.httpServer === null) {
    throw new Error('HTTP server config not defined')
}
```

Maybe the same logic could be applied here? Note also that in https://github.com/streamr-dev/network-monorepo/pull/145 we'll add a default port -> a `httpServer` config will always be available.